src/Microsoft.Security.Utilities.Core/HighPerformanceScanner.cs

```diff
@@ -28,25 +29,48 @@ namespace Microsoft.Security.Utilities;
 /// </summary>
 internal sealed class HighPerformanceScanner
 {
+    private Dictionary<int, CompiledHighPerformancePattern> _patternsByPackedSignature = new();
+
 #if NET9_0_OR_GREATER
-    private SearchValues<string> _signatures;
+    private static SearchValues<string> s_emptySignatures = SearchValues.Create(Array.Empty<string>(), StringComparison.Ordinal);
+    private SearchValues<string> _signatures = s_emptySignatures;
 #elif NET8_0_OR_GREATER
-    private SearchValues<char> _signatureStarts;
+    private static SearchValues<char> s_emptySignatureStarts = SearchValues.Create(Array.Empty<char>());
```

Comment:
>Create [](http://example.com/codeflow?start=76&length=6)

SEV3: my assumption here is that this code path does not support adding the `StringComparison.Ordinal` argument?

Reply:
Actually, yes, there is no StringComparison for a search value of chars. Of course. But still, sentinel empty search values was bad. Fixed.